### ISSUE TITLE
Cleanup get_partition_id_from_root

### ISSUE THREAD

**Issue by smarlowucf** [SUSE-Enceladus/rootgrow#9]
As there have been two regression patches to the function it could be cleaned up to be more readable.

Ensure the function still works for both SSD and NVME drives with id's like /dev/sda1 and /dev/nvme0n1p3. Also, there's no longer a guarantee that the partitions are ordered by number.

**Comment by smarlowucf** [comment]
Currently the main function uses the root_device to find the root_device from the root_disk. This seems unnecessary as we already have the root_device. Suggest simplifying this to:

main function
```
root_device = get_root_device()  # Here we get the root_device
    if root_device:
        root_fs = get_root_filesystem_type(root_device)
        root_disk = get_disk_device_from_root(root_device)
        root_part = get_partition_id_from_root(root_device)  # Here we are getting the partition number from the root device
```

get part id
```
def get_partition_id_from_root(root_device):
    partition_index = re.match('.*?([0-9]+)$', root_device).group(1)
    return int(partition_index)
```

As I don't fully understand the use case maybe I'm missing something here? Is this suggestion workable? Also, I'm not sure if this function (get_partition_id_from_root) is being used anywhere else outside of this code.

**Comment by smarlowucf** [comment]
@schaefi @rjschwei @jgleissner thoughts here?

**Comment by rjschwei** [comment]
Solved with #10
